tim/opensearch.py

```diff
@@ -101,6 +104,36 @@ def get_formatted_indexes(client: OpenSearch) -> str:
     return "\nNo indexes present in OpenSearch cluster.\n"
 
 
+def get_all_aliased_indexes_for_source(
+    client: OpenSearch, source: str
+) -> Optional[dict[str, list[str]]]:
+    """Return all aliased indexes for the source, grouped by alias.
+
+    Returns a dict of the aliases with a list of the source index(es) for each. There
+    *should* only be one source index per alias, but in case there is ever more than
+    one, this function will return the accurate information and log an error for
+    further investigation.
+
+    Returns None if there are no aliased indexes for the source.
+    """
+    result = {}
+    if aliases := get_aliases(client):
+        logger.debug(aliases)
+        for alias, indexes in aliases.items():
+            source_indexes = [
+                index for index in indexes if index.split("-")[0] == source
```

Comment:
Yes, TIM will enforce that pattern during index creation. If there's no `-` in the index name, `index.split("-")` will just return the whole index name so this will evaluate to false. I intentionally used `string.split()` instead of `string.startswith()` in case of the unlikely but not impossible scenario where we have a source whose name starts with the name of another source, e.g. `dspace` and `dspacetwo`.